framework/src/main/java/org/fulib/fx/constructs/Modals.java

```diff
@@ -0,0 +1,226 @@
+         * Adds a pre-made initializer with some default options regarding transparency and modality.
+         *
+         * @param dialog Whether the default dialog options should be used
+         * @return The current modal instance
+         */
+        public Modal<T> dialog(boolean dialog) {
+            this.dialog = dialog;
+            return this;
+        }
+
+        /**
+         * Sets the parameters to be used when initializing/rendering the component.
+         *
+         * @param params The parameter map
+         * @return The current modal instance
+         */
+        public Modal<T> params(Map<String, Object> params) {
+            this.params = params;
+            return this;
+        }
+
+        /**
+         * Whether the component should be destroyed when the modal is closed.
+         *
+         * @param destroyOnClose Whether the component should be destroyed when the modal is closed
+         * @return The current modal instance
+         */
+        public Modal<T> destroyOnClose(boolean destroyOnClose) {
+            this.destroyOnClose = destroyOnClose;
+            return this;
+        }
+
+        /**
+         * Displays the current modal.
+         * This can only be called once per modal builder.
+         */
+        public void show() {
+
+            if (component.getScene() != null) {
+                throw new RuntimeException(error(1014));
+            }
+
+            FulibFxApp.FX_SCHEDULER.scheduleDirect(() -> {
+                ModalStage modalStage = new ModalStage(app, destroyOnClose, component);
+
+                // Add additional default parameters
+                Map<String, Object> parameters = params == null ? new HashMap<>() : new HashMap<>(params);
+                parameters.putIfAbsent("modalStage", modalStage);
+                parameters.putIfAbsent("ownerStage", owner);
+
+                // Initialize and render the component
+                app.frameworkComponent().controllerManager().init(component, parameters);
+                Node rendered = app.frameworkComponent().controllerManager().render(component, parameters);
+
+                // As the displayed component will be the root of a stage, it has to be a parent
+                if (!(rendered instanceof Parent parent)) {
+                    throw new IllegalArgumentException(error(1011).formatted(component.getClass().getName()));
+                }
+
+                // Set the title if present
+                app.applyTitle(component, modalStage);
+
+                // Setup the stage and scene
+                Scene scene = new Scene(parent);
+                modalStage.setScene(scene);
+                modalStage.initOwner(owner);
+                if (dialog) {
+                    FULIBFX_DIALOG.accept(modalStage, component);
+                }
+                if (initializer != null) {
+                    initializer.accept(modalStage, component);
+                }
+                modalStage.show();
```

Comment:
While the initializer is good, we should probably still offer a way to build the stage and do something before show. i.e split this method here into build() and show(). Both are terminal operations for the builder.